spyder/plugins/editor/widgets/editor.py

```diff
@@ -1799,7 +1799,7 @@ def compute_hash(self, fileinfo):
         Returns:
             int: computed hash.
         """
-        txt = fileinfo.editor.get_text_with_eol()
+        txt = to_text_string(fileinfo.editor.get_text_with_eol())

```

Comment:
Is this really necessary? I mean, why do you need to convert the file contents to unicode before getting their hash?

Reply:
As I said in https://github.com/spyder-ide/spyder/pull/15903#discussion_r665448937, I converted the file contents to Unicode to keep the original behavior of ``compute_hash``. If this also shouldn't be doing, I can remove it, but I still don't know if this change should be done in this PR.